### DISCUSSION TITLE
No documentation on how to "Sign in" for the first time

### DISCUSSION THREAD

**Discussion by DerReparator** [teslamate-org/teslamate#3521] [Q&A] [answered]
Hello,

I followed the instructions [here](https://docs.teslamate.org/docs/installation/debian/) to install TeslaMate in a Debian 12 LXC under Proxmox 8.1.3.

But I cannot find any documentation on how to setup TeslaMate now? My page looks like this:

![image](https://github.com/teslamate-org/teslamate/assets/30900810/6626b174-5a0b-4447-82bf-e145df3d9bb9)

But when I enter my tokens created with _Auth for Tesla_ (iOS), the _Sign In_ button still doesn't work.
The Tokens from this iOS app are working with other Tesla API services.

Using `journalctl -u teslamate -f` in my LXC container, I recognized the following entry whenever I access `http://teslamate.fritz.box:4000/sign_in`:

```
Nov 29 21:55:44 teslamate teslamate[61583]: 2023-11-29 22:55:44.544 [info] GET /
Nov 29 21:55:44 teslamate teslamate[61583]: 2023-11-29 22:55:44.547 [info] Sent 302 in 2ms
Nov 29 21:55:44 teslamate teslamate[61583]: 2023-11-29 22:55:44.556 [info] GET /sign_in
Nov 29 21:55:44 teslamate teslamate[61583]: 2023-11-29 22:55:44.559 [info] Sent 200 in 2ms
Nov 29 21:55:44 teslamate teslamate[61583]: 2023-11-29 22:55:44.617 [info] GET /assets/app.css
Nov 29 21:55:44 teslamate teslamate[61583]: 2023-11-29 22:55:44.617 [info] Sent 404 in 61µs
Nov 29 21:55:44 teslamate teslamate[61583]: 2023-11-29 22:55:44.619 [info] GET /assets/app.js
Nov 29 21:55:44 teslamate teslamate[61583]: 2023-11-29 22:55:44.619 [info] Sent 404 in 41µs
``` 

Is the `404` code something to worry about?

Can you help me with finishing my setup?

**Comment by cwanja** [comment]
`404` means it cannot find files. The .css is a stylesheet and might be why you are not seeing styling on the webpage.

It seems like your install is missing files or is broken.

**Reply by DerReparator** [reply]
That makes sense. The `assets/css/app.scss` did not get compiled it seems - I guess that this would have produced the `app.css` file...

Should I _Compile Elixir_ again? Or which step would create these files?

Edit:

During re-compile, Elixir mentions that my VM is still running _latin1_:

```
warning: the VM is running with native name encoding of latin1 which may cause Elixir to malfunction as it expects utf8. Please ensure your locale is set to UTF-8 (which can be verified by running "locale" in your shell) or set the ELIXIR_ERL_OPTIONS="+fnu" environment variable
```

I am not able to change this behavior. I set the env variable `ELIXIR_ERL_OPTIONS`, I executed the code from the docs again: 
```
sudo locale-gen en_US.UTF-8
sudo localectl set-locale LANG=en_US.UTF-8
```

I verified the `LANG` variable in `/etc/default/locale`. Everything is set to _UTF-8_ but when I execute the compile it says, the language is set to _latin1_.

**Reply by cwanja** [reply]
To be frank, I am not experienced enough in the manual install. Adding the tag to request help. Repo is more active now, hopefully someone will pick up the discussion and guide you. 

My poor poor advice - if you have no data, nuke the install and start completely over.

**Comment by brianmay** [comment, marked as answer]
Procedure varies depending on if you are running prod or dev mode.

I think in dev mode the following is sufficient, and restart phx server. I believe this should automatically build the assets.

```
npm install --prefix ./assets
```

For prod mode see the instructions. https://docs.teslamate.org/docs/installation/debian#compile-elixir-project

**Reply by DerReparator** [reply]
That was indeed the answer.

I forced my locale to UTF-8.

Then I re-did the steps to _Compile the Elixir project_. I can now login to my TeslaMate instance. Thank you!

**Reply by cwanja** [reply]
Thanks @brianmay and glad to hear it is up and running @DerReparator. Apologize for removing your marked answered, wrong button press. Remarked as answered and closed.
